### PR TITLE
Add symbols, computed props to effects handlers

### LINKED ISSUE #7: Add symbols, computed props to effects handlers
## Problem

As of right now, effect handlers can only be stringly / numberly typed.

## Solution

Alter the transform plugin to correctly interpolate and expect Symbols as well as computed properties in effects handlers.

## Acceptance Criteria

The following code samples should transpile

```
const {computedType} = require('@effects/type');
const symbolType = Symbol('effect');

'use effects'
try{
  perform {type : symbolType};
  perform {type : computedType};
}handle(e}{
  switch(e.type){
    case symbolType : return symbolTypeHandler();
    case computedType: return computedTypeHandler();
  }
}
```

### PR DESCRIPTION
**Pull request by dpchamps**
closes #7

## problem

Effect handler types were limited to locally-declared, stringly-typed variables. 

## solution

Preform the transformation more intelligently, allow for computed properties on the handler object.

### CODE REVIEW

File: packages/babel-plugin-effects/src/handler-method-visitor.ts [hunk 1]
```diff
@@ -23,27 +27,33 @@ const isCorrectMemberPropertyPath = (
...
 
-    return extractMemberPropertyPathName(
-      binding.path.parent,
-      types,
-      binding?.path.node.init
-    );
+    // I think at this point it's safe to throw.
   }
 
   throw new Error(
```
Comment (cdaringe) [inline packages/babel-plugin-effects/src/handler-method-visitor.ts:59]: you did it!

File: packages/effects-common/src/effects.types.ts [hunk 2]
```diff
@@ -10,7 +10,9 @@ export interface Handler {
   [index: number]: HandlerDefinition;
 }
 
+export type HandlerType = string | number | symbol;
+
 export interface Effect {
-  type: string | number;
+  type: HandlerType;
```
Comment (cdaringe) [inline packages/effects-common/src/effects.types.ts]: 🌶

File: packages/effects-common/src/mod.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export class UnhandledEffectError extends TypeError {
   constructor({ type }: Effect) {
     super();
 
-    this.message = `Encountered an unhandled effect :${type}`;
+    this.message = `Encountered an unhandled effect :${String(type)}`;
```
Comment (cdaringe) [inline packages/effects-common/src/mod.ts:25]: huh.

```
[0,null, undefined, 'str', NaN, 1e20, true, false, Infinity, Symbol("wut")].map(thing => ([String(thing)]))
```
works, but not in string templates.  _weirt_.

File: packages/effects-runtime/test/__fixtures__/functional-tests/computed-handlers.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    main(() => {
+        return perform {type : computedHandler}
+    })
+};
+
+const locallyScopedEffects = async () => {
+  'use effects';
+  const something = 'something';
+  try{
+      return perform {type : 'something'};
```
Comment (cdaringe) [inline packages/effects-runtime/test/__fixtures__/functional-tests/computed-handlers.js]: why the literal here?

File: packages/effects-runtime/test/effects.spec.ts [hunk 5]
```diff
@@ -350,5 +350,36 @@ describe("Effects Unit Tests", () => {
 
       stackResume(controlFrame);
     });
+
+    it("Should handle symbol-y typed effect-handles", async () => {
```
Comment (cdaringe) [inline packages/effects-runtime/test/effects.spec.ts:354]: lol. ❤️

**Review by cdaringe — APPROVED** [review]

File: packages/effects-common/src/mod.ts (hunk 3 above)
Comment (dpchamps) [reply packages/effects-common/src/mod.ts:25]: Yeah. Super annoying.

File: packages/effects-common/src/effects.types.ts (hunk 2 above)
Comment (dpchamps) [reply packages/effects-common/src/effects.types.ts]: Hmmm.. that probably should be EffectType looking at it again!

File: packages/effects-runtime/test/__fixtures__/functional-tests/computed-handlers.js (hunk 4 above)
Comment (dpchamps) [reply packages/effects-runtime/test/__fixtures__/functional-tests/computed-handlers.js]: good catch!

**Pull request merged**